plugin.rb

```diff
@@ -146,6 +146,23 @@ def self.allowed_group_ids
     results
   end
 
+  if respond_to?(:register_upload_unused_callback)
+    register_upload_unused_callback do |uploads|
+      uploads
+        .joins("LEFT JOIN chat_uploads pu ON cu.upload_id = uploads.id")
+        .where("cu.upload_id IS NULL")
+    end
+  end
+
+  if respond_to?(:register_upload_in_use)
+    register_upload_in_use do |upload|
+      # TODO after May 2022 - remove this. No longer needed as chat uploads are in a table
+      next true if ChatMessage.where("message LIKE ? OR message LIKE ?", "%#{upload.sha1}%", "%#{upload.base62_sha1}%").exists?
+
+      false
```

Comment:
Couldn't this be a one-liner instead?

```suggestion
      ChatMessage..exists?("message LIKE ? OR message LIKE ?", "%#{upload.sha1}%", "%#{upload.base62_sha1}%")
```